tsMuxer/blurayHelper.cpp

```diff
@@ -165,22 +166,29 @@ bool BlurayHelper::writeBluRayFiles(bool usedBlackPL, int mplsNum, int blankNum,
         delete file;
         return false;
     }
-
+    uint8_t* emptyCommand;
     if (m_dt == DT_BLURAY) {
         bdMovieObjectData[5] = bdIndexData[5] = '2';
         fileSize = 0x78;
     }
     else if (m_dt == UHD_BLURAY) {
         bdMovieObjectData[5] = bdIndexData[5] = '3';
-        fileSize = 0x78;
+        fileSize = 0x9C; // add 36 bytes for HDR data extension
+        bdIndexData[15] = 0x78; // start address of HDR data extension
+        emptyCommand = bdIndexData + 0x78;
+        memcpy(emptyCommand, "\x00\x00\x00\x20\x00\x00\x00\x18\x00\x00\x00\x01"
+                             "\x00\x03\x00\x01\x00\x00\x00\x18\x00\x00\x00\x0C"
+                             "\x00\x00\x00\x08\x51\x00\x00\x00\x00\x00\x00\x00", 36); // HDR data extension

```

Comment:
If you replace this string literal with an array of `uint8_t`, you'll get less code and will be able to use `sizeof` instead of hard-coding the size of the array.